tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/CiamIntegrationTests.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            //Get lab details
+            var labResponse = await LabUserHelper.GetLabUserDataAsync(new UserQuery() 
+            { 
+                FederationProvider = FederationProvider.CIAM,
+                SignInAudience = SignInAudience.AzureAdMyOrg,
+                PublicClient = PublicClient.no
+            }).ConfigureAwait(false);
+
+            string authority = string.Empty;
+
+            //Compute authority from format and lab response
+            switch (authorityVersion)
+            {
+                case 0:
+                    authority = string.Format(authorityFormat, labResponse.User.LabName);
+                    break;
+
+                case 1:
+                    authority = string.Format(authorityFormat, labResponse.User.LabName, labResponse.User.LabName);
+                    break;
+
+                case 2:
+                    authority = string.Format(authorityFormat, labResponse.User.LabName, labResponse.Lab.TenantId);
+                    break;
+            }
+
+            //Acquire tokens
+            var msalPublicClient = PublicClientApplicationBuilder
+                .Create(labResponse.App.AppId)
+                .WithAuthority(authority, false)
+                .WithRedirectUri(_ciamRedirectUri)
+                .Build();
+
+            var result = await msalPublicClient
+                .AcquireTokenByUsernamePassword(_ciamScopes, labResponse.User.Upn, labResponse.User.GetOrFetchPassword())
+                .WithExtraQueryParameters(_extraQParams)
+                .ExecuteAsync()
+                .ConfigureAwait(false);
+
+            Assert.IsNotNull(result.AccessToken);
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+            Assert.AreEqual($"{labResponse.User.LabName}{Constants.CiamAuthorityHostSuffix}".ToLower(), result.Account.Environment);
+
+            //Fetch cached tokens
+            var accounts = await msalPublicClient.GetAccountsAsync().ConfigureAwait(false);
+
+            result = await msalPublicClient
+                .AcquireTokenSilent(_ciamScopes, accounts.FirstOrDefault())
+                .ExecuteAsync()
+                .ConfigureAwait(false);
+
+            Assert.IsNotNull(result.AccessToken);
+            Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+            Assert.AreEqual($"{labResponse.User.LabName}{Constants.CiamAuthorityHostSuffix}".ToLower(), result.Account.Environment);
+        }
+
+        [TestMethod]
+        [DataRow("https://{0}.ciamlogin.com/", 0)] //https://tenantName.ciamlogin.com/
+        [DataRow("https://{0}.ciamlogin.com/{1}.onmicrosoft.com", 1)] //https://tenantName.ciamlogin.com/tenantName.onmicrosoft.com
+        [DataRow("https://{0}.ciamlogin.com/{1}", 2)] //https://tenantName.ciamlogin.com/tenantId
+        public async Task ClientCredentialWithClientSecret_Ciam_Async(string authorityFormat, int authorityVersion)
```

Comment:
I didnt see a certificate anywhere to use. I will Ask Ryan if it is available so we can also test with the cert.

Reply:
You can also reach out to @siddhijain - she has managed to add E2E tests for client_credentials. 
Also, it seems that acquire token interactive is pretty simple to automate as well. 

But this can be done separately.